test/e2e/main/main_test.go

```diff
@@ -124,25 +124,83 @@ func TestNodeStartup(t *testing.T) {
 	pod := corev1.Pod{}
 	tutils.ExpectNoError(testKube.Kubernetes.Client.Get(context.TODO(), types.NamespacedName{Name: spec.Name, Namespace: tutils.Namespace}, &ispn))
 	tutils.ExpectNoError(testKube.Kubernetes.Client.Get(context.TODO(), types.NamespacedName{Name: spec.Name + "-0", Namespace: tutils.Namespace}, &pod))
-	if pod.Labels["my-pod-label"] != ispn.Labels["my-pod-label"] ||
-		pod.Labels["operator-pod-label"] != "operator-pod-value" ||
-		ispn.Labels["operator-pod-label"] != "operator-pod-value" {
-		panic("Labels haven't been propagated to pods")
+
+	// Checking labels propagation to pods
+	// from Infinispan CR to pods
+	if pod.Labels["my-pod-label"] != ispn.Labels["my-pod-label"] {
+		panic("Infinispan CR labels haven't been propagated to pods")
+	}
+
+	// from operator environment
+	if tutils.RunLocalOperator == "TRUE" {
+		// running locally, labels are hardcoded and set by the testsuite
+		if pod.Labels["operator-pod-label"] != "operator-pod-value" ||
+			ispn.Labels["operator-pod-label"] != "operator-pod-value" {
+			panic("Infinispan CR labels haven't been propagated to pods")
+		}
+	} else {
+		// operator deployed on cluster, labels are set by the deployment
+		if !areOperatorLabelsPropagated(spec.Namespace, ispnv1.OperatorPodTargetLabelsEnvVarName, pod.Labels) {
```

Comment:
Operator doesn't necessarily have to run in the same namespace as the Infinispan cluster. Can we parameterize expected Operator namespace?

Reply:
@Crumby I've added a commit for this: operator NS can be configured via `TESTING_OPERATOR_NAMESPACE` envvar